### PR TITLE
feat: remove patching jaeger exporter

### PR DESCRIPTION
**Pull request by rauno56**
# Description

We used to patch Jaeger exporter to implement attribute limits. We don't need that anymore since it's implmemented in OTel now.

## Type of change

- [x] Internal change (a change which is not visible to the package consumers)
- [x] Tested manually
- [x] Documentation has been updated
- [x] Change file has been generated (`npm run change:new`)
- [x] Delete this branch (after the PR is merged)

### CODE REVIEW

File: docs/advanced-config.md [hunk 1]
```diff
@@ -32,7 +32,6 @@ The following table contain the configuration options supported by this distribu
 | OTEL_PROPAGATORS                     | propagators                     | `tracecontext,baggage`                | Comma-delimited list of propagators to use. Valid keys: `baggage`, `tracecontext`, `b3multi`, `b3`.
 | OTEL_SERVICE_NAME                    | serviceName                     | `unnamed-node-service`                | The service name of this Node service.
 | SPLUNK_ACCESS_TOKEN                  | accessToken                     |                                       | The optional access token for exporting signal data directly to SignalFx API.
-| OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT | maxAttrLength                 | 1200                                  | Maximum length of string attribute value in characters. Longer values are truncated.
```
Comment (owais) [inline docs/advanced-config.md:35]: We shouldn't remove documentation for this as we are setting this to non-otel default.

Comment (Rauno56) [reply docs/advanced-config.md:35]: We're setting non-otel default to `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT`(very similar name, but more general) and that is documented :point_up: on line 23 here.

File: src/options.ts [hunk 2]
```diff
@@ -57,7 +56,6 @@ export interface Options {
   endpoint?: string;
   serviceName: string;
   accessToken: string;
-  maxAttrLength: number;
```
[the remark this answers is not in the document]
Comment (Rauno56) [reply src/options.ts:60]: I'd rather remove that and leave only the most essential config options for programmatic APIs because:

1. Having those provides very little value but has overhead;
2. There's a lot of those missing options and relationships between them - example of `OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT` and `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT` is one of them;
3. In an extreme vase it can be mimicked bit just setting `process.env` properties.

In conclusion: If we should allow this, why shouldn't we implement the rest of the limits + the general limits?

[the remark this answers is not in the document]
Comment (owais) [reply src/options.ts:60]: OK fine with removing and allowing only env var based config for limits

**Review by owais — APPROVED** [review]
